plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,342 @@
+    except Exception as err:
+        msg = "Failed to get record of file %s with error: %s" % (dest_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    try:
+        if ISO_MODE == "iso9660":
+            opened_iso.rm_file(iso_path=file_in_iso_path)
+        elif ISO_MODE == "rr":
+            opened_iso.rm_file(iso_path=file_in_iso_path, rr_name=os.path.basename(dest_file))
+        elif ISO_MODE == "joliet":
+            opened_iso.rm_file(joliet_path=dest_file)
+        elif ISO_MODE == "udf":
+            opened_iso.rm_file(udf_path=dest_file)
+    except Exception as err:
+        msg = "Failed to delete iso file %s with error: %s" % (dest_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return 0, ""
+
+
+def iso_rebuild(src_iso, dest_iso, op_data_list):
+    global ISO_MODE
+
+    try:
+        iso = pycdlib.PyCdlib()
+        iso.open(src_iso)
+        if iso.has_rock_ridge():
+            ISO_MODE = "rr"
+        elif iso.has_joliet():
+            ISO_MODE = "joliet"
+        elif iso.has_udf():
+            ISO_MODE = "udf"
+
+        for op_data in op_data_list:
+            op = op_data['op'].strip()
+            data = op_data['data']
+            if op == "delete_files":
+                for item in data:
+                    ret, msg = iso_delete_file(iso, item.strip())
+                    if ret != 0:
+                        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+            elif op == "add_files":
+                for item in data:
+                    ret, msg = iso_add_file(iso, item['src_file'],
+                                            item['dest_file'])
+                    if ret != 0:
+                        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+        iso.write(dest_iso)
+    except Exception as err:
+        msg = "Failed to rebuild ISO %s with error: %s" % (src_iso, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+    finally:
+        if iso:
+            iso.close()
+    return
+
+
+def main():
+    op_data_list = []
+    op_data_item = {}
+    global MODULE_ISO_CUSTOMIZE
+
+    argument_spec = dict(
+        src_iso=dict(type='str', required=True),
+        dest_iso=dict(type='str', required=True),
+        delete_files=dict(type='list', required=False, elements='str'),
+        add_files=dict(type='list', required=False, elements='dict'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+    )
+    if not HAS_PYCDLIB:
+        module.fail_json(
+            missing_required_lib('pycdlib'), exception=PYCDLIB_IMP_ERR)
+
+    src_iso = module.params.get('src_iso')
+    if not os.path.exists(src_iso):
+        module.fail_json(msg="The %s does not exist." % src_iso)
+
+    dest_iso = module.params.get('dest_iso')
+    if not dest_iso:
+        module.fail_json(str(msg='Please specify the path of the customized ISO file \
+            using dest_iso parameter.'))
```

Comment:
100? I though the limit was 160? @felixfontein ?

Reply:
I mark this issue as "Resolved" in order to remove label "need_revision" since there is no response.
Please reopen it if it still needs be changed.

Thanks very much.

Reply:
no response for a few days and close this conversation